File: services/src/main/java/com/liferay/mobile/sdk/v7/app/AppService.java
package com.liferay.mobile.sdk.v7.app;

import com.liferay.mobile.sdk.Call;
import com.liferay.mobile.sdk.annotation.Param;
import com.liferay.mobile.sdk.annotation.Path;
import com.liferay.mobile.sdk.file.UploadData;
import com.liferay.mobile.sdk.http.ContentType;
import com.liferay.mobile.sdk.http.Response;
import org.json.JSONObject;

@Path("/marketplace.app")
public interface AppService {
  @Path("/delete-app")
  Call<JSONObject> deleteApp(@Param(name = "appId") long appId);

  @Path("/install-app")
  Call<Response> installApp(@Param(name = "remoteAppId") long remoteAppId);

  @Path("/uninstall-app")
  Call<Response> uninstallApp(@Param(name = "remoteAppId") long remoteAppId);

  @Path(
      value = "/update-app",
      contentType = ContentType.MULTIPART
  )
  Call<JSONObject> updateApp(@Param(name = "file") UploadData file);
}

File: windows/Liferay.SDK/Service/V73/App/AppService.cs
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Liferay.SDK.Service.V73.App
{
	public class AppService : ServiceBase
	{
		public AppService(ISession session)
			: base(session)
		{
		}

		public async Task<dynamic> DeleteAppAsync(long appId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("appId", appId);

			var _command = new JsonObject()
			{
				{ "/marketplace.app/delete-app", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> UpdateAppAsync(Stream file)
		{
			var _parameters = new JsonObject();

			_parameters.Add("file", file);

			var _command = new JsonObject()
			{
				{ "/marketplace.app/update-app", _parameters }
			};

			var _obj = await this.Session.UploadAsync(_command);

			return (dynamic)_obj;
		}

		public async Task InstallAppAsync(long remoteAppId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("remoteAppId", remoteAppId);

			var _command = new JsonObject()
			{
				{ "/marketplace.app/install-app", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}

		public async Task UninstallAppAsync(long remoteAppId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("remoteAppId", remoteAppId);

			var _command = new JsonObject()
			{
				{ "/marketplace.app/uninstall-app", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}
	}
}